### ISSUE TITLE
RUF100 only removes noqa string when using names

### ISSUE THREAD

**Issue by JannKleen** [charliermarsh/ruff#4664]
When having the following code:
```python
def test():  # noqa: too-many-branches
  pass
```

RUF100 would only remove the noqa, leaving this:
```python
def test():  # : too-many-branches
  pass
```

Edit:
The removal of the 'numeric' codes (i.e. `# noqa: PLR0912`)  works as expected.

**Comment by JonathanPlasse** [comment]
- `noqa`'s with rule names is not supported (c.f. #1773)

**Comment by charliermarsh** [comment]
We don't support this right now, but we likely will in the future.
